### PR TITLE
migrate `/etc/noteable` to `/var/run/noteable` with symlink

### PR DESCRIPTION
**Pull request by shouples**
## Describe your changes

Sets `NBL_WORK_DIR` as a build arg (default `/var/run/noteable`) and moves main workdir to `/var/run/noteable/project`, removing `/etc/noteable` altogether and symlinking while other services migrate and kernels change over. 

Sets `/srv/noteable` as the `NBL_HOME_DIR` build arg.

This will be followed-up with another PR to remove the symlink and `/etc/noteable` altogether.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I am able to build images locally
- [x] Has the issue it resolves been discussed with maintainers?

### CODE REVIEW

File: deno/base/1.37/Dockerfile [hunk 1]
```diff
@@ -6,25 +6,29 @@
 # ---
 # Deno is installed from a PR that adds Jupyter support
 # Rust 1.72 is required
-ARG NBL_DENO_VERSION="1.37-dev"
```
Comment (shouples) [inline deno/base/1.37/Dockerfile:9]: unused

File: python/base/3.10/Dockerfile [hunk 2]
```diff
@@ -7,6 +7,11 @@
 ARG NBL_PYTHON_VERSION=3.10
 FROM python:${NBL_PYTHON_VERSION}-slim-bullseye as base
 
+# Application code lives in NBL_HOME_DIR, project data lives in NBL_WORK_DIR
+ARG NBL_HOME_DIR=/srv/noteable
+ARG NBL_WORK_DIR=/var/run/noteable
+ARG NBL_PYTHON_VERSION
```
Comment (tizz98) [inline python/base/3.10/Dockerfile:13]: Is this a duplicate of the above `ARG`?

File: python/base/3.11/Dockerfile [hunk 3]
```diff
@@ -7,6 +7,11 @@
 ARG NBL_PYTHON_VERSION=3.11
 FROM python:${NBL_PYTHON_VERSION}-slim-bullseye as base
 
+# Application code lives in NBL_HOME_DIR, project data lives in NBL_WORK_DIR
+ARG NBL_HOME_DIR=/srv/noteable
+ARG NBL_WORK_DIR=/var/run/noteable
+ARG NBL_PYTHON_VERSION
```
Comment (tizz98) [inline python/base/3.11/Dockerfile:13]: duplicate here too?

File: python/base/3.9/Dockerfile [hunk 4]
```diff
@@ -7,6 +7,11 @@
 ARG NBL_PYTHON_VERSION=3.9
 FROM python:${NBL_PYTHON_VERSION}-slim-bullseye as base
 
+# Application code lives in NBL_HOME_DIR, project data lives in NBL_WORK_DIR
+ARG NBL_HOME_DIR=/srv/noteable
+ARG NBL_WORK_DIR=/var/run/noteable
+ARG NBL_PYTHON_VERSION
```
Comment (tizz98) [inline python/base/3.9/Dockerfile:13]: here too?

File: Taskfile.R.yaml [hunk 5]
```diff
@@ -26,7 +26,7 @@ tasks:
     cmds:
       - task r:base:copy-files IDENTIFIER=base NBL_LANGUAGE_VERSION={{.NBL_R_VERSION}}
       # build the base image
-      - task build LANGUAGE=r NBL_LANGUAGE_VERSION={{.NBL_R_VERSION}} IDENTIFIER=base -- --build-arg NBL_R_VERSION={{.NBL_R_VERSION}}
+      - task build LANGUAGE=r NBL_LANGUAGE_VERSION={{.NBL_R_VERSION}} IDENTIFIER=base BUILD_TARGET=base -- --build-arg NBL_R_VERSION={{.NBL_R_VERSION}}
```
Comment (shouples) [inline Taskfile.R.yaml:29]: this was stuck looking at `BUILD_TARGET=main` for some reason; should be cleared up in https://github.com/noteable-io/kernels/pull/59 and only affects local building here

**Review by tizz98 — APPROVED** [review]

**Pull request merged**